universe/supplycommit/supply_commit_states.go

```diff
@@ -0,0 +1,553 @@
+func (c *CommitTxCreateState) String() string {
+	return "CommitTxCreateState"
+}
+
+// SignTxEvent is a special event that is used to trigger the signing of the
+// commitment transaction.
+type SignTxEvent struct {
+	// CommitPkt is the unsigned transaction that will be used to commit to
+	// the new supply tree.
+	CommitPkt *tapsend.FundedPsbt
+
+	// NewSupplyCommit is the new supply commitment that that will be
+	// created by the above commit pkt.
+	NewSupplyCommit RootCommitment
+}
+
+// eventSealed is a special method that is used to seal the interface.
+func (s *SignTxEvent) eventSealed() {}
+
+// CommitTxSignState is the state of the state machine we'll transition to once
+// we've created the tx that commits to the latest supply tree. In this state,
+// we'll sign the transaction and all its inputs, then write our state to disk
+// so we'll be able to recover it on restart.
+//
+// State transitions:
+//   - SignTxEvent -> CommitBroadcastState
+type CommitTxSignState struct {
+	// SupplyTransition holds all the information about the current state
+	// transition, including old/new trees and commitments.
+	SupplyTransition SupplyStateTransition
+}
+
+// stateSealed is a special method that is used to seal the interface.
+func (s *CommitTxSignState) stateSealed() {}
+
+// IsTerminal returns true if the target state is a terminal state.
+func (s *CommitTxSignState) IsTerminal() bool {
+	return false
+}
+
+// String returns the name of the state.
+func (s *CommitTxSignState) String() string {
+	return "CommitTxSignState"
+}
+
+// BroadcastEvent is a special event that is used to trigger the broadcasting of
+// the commitment transaction.
+type BroadcastEvent struct {
+	// SignedCommitPkt is the signed commitment transaction that will be
+	// broadcast to the network.
+	SignedCommitPkt *psbt.Packet
+}
+
+// eventSealed is a special method that is used to seal the interface.
+func (b *BroadcastEvent) eventSealed() {}
+
+// ConfEvent is a special event sent once our latest commitment transaction
+// confirms on chain.
+type ConfEvent struct {
+	// BlockHeight is the height of the block in which the transaction was
+	// confirmed within.
+	BlockHeight uint32
+
+	// TxIndex is the index within the block of the ultimate confirmed
+	// transaction.
+	TxIndex uint32
+
+	// Tx is the transaction for which the notification was requested for.
+	Tx *wire.MsgTx
+
+	// Block is the block that contains the transaction referenced above.
+	Block *wire.MsgBlock
+}
+
+// eventSealed is a special method that is used to seal the interface.
+func (c *ConfEvent) eventSealed() {}
+
+// CommitBroadcastState is the state of the state machine we'll transitions
+// to once we've signed the transaction. In this state, we'll broadcast the
+// transaction, then wait for a confirmation event.
+//
+// State transitions:
+//   - BroadcastEvent -> CommitBroadcastState
+//   - ConfEvent -> DefaultState
+type CommitBroadcastState struct {
```

Comment:
I don't understand `ConfEvent -> DefaultState` in the doc here.

Once our latest commitment transaction confirms on chain the state machine could transition to `DefaultState`?

Reply:
This is slightly out of date, a new `CommitFinalizeState` state is was transitions us to the default.